google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineAssetProvider.java

```diff
@@ -44,4 +46,25 @@ public XmlFile loadAppEngineStandardWebXml(@NotNull Project project, @NotNull Ar
 
     return null;
   }
+
+  @Nullable
+  @Override
+  public XmlFile loadAppEngineStandardWebXml(@NotNull Project project, @NotNull Module module) {
+    VirtualFile[] contentRoots = ModuleRootManager.getInstance(module).getContentRoots();
+
+    for (VirtualFile contentRoot : contentRoots) {
+      // TODO clean this up - do we need something more robust?
+      VirtualFile descriptorFile
+          = contentRoot.findFileByRelativePath("web/WEB-INF/appengine-web.xml");
```

Comment:
The old plugin did this at the artifact level - similarly to how I'm doing it in the overloaded method AppEngineAssetProvider.

For these changes, I needed to do this at the module level. With some discussion with JB on the forums I found no good way to do this yet at the module level (https://intellij-support.jetbrains.com/hc/en-us/community/posts/208391445-How-to-get-a-PsiFile-from-a-Module-).